ext/afform/core/afform.php

```diff
@@ -651,3 +651,16 @@ function afform_shortcode_content($content, $atts, $args, $context) {
   }
   return $content;
 }
+
+/**
+ * Implements hook_civicrm_searchKitTasks().
+ *
+ */
+function afform_civicrm_searchKitTasks(array &$tasks, bool $checkPermissions, ?int $userID) {
+  $tasks['AfformSubmission']['process'] = [
+    'module' => 'afSearchTasks',
+    'title' => E::ts('Process Submissions'),
+    'icon' => 'fa-check-square-o',
+    'uiDialog' => ['templateUrl' => '~/afSearchTasks/afformSubmissionProcessTask.html'],
```

Comment:
@kurund looking at the code in this `afformSubmissionProcessTask.html` and the related angular controllers/modules, it looks to me like all it does is make one api call with no user-configurable params. In that case, I think this could all be simplified as an `apiBatch`. You could delete the new angular module/js/html and just replace this line with:
```suggestion
        'apiBatch' => [
          'action' => 'process',
          'params' => NULL,
          'confirmMsg' => E::ts('Confirm processing %1 %2?'),
          'runMsg' => E::ts('Processing %1 %2...'),
          'successMsg' => E::ts('Successfully processed %1 %2.'),
          'errorMsg' => E::ts('An error occurred while attempting to process %1 %2.'),
        ],
```

Hmm, but now that I look at it, it seems the `process` action has been added to the Afform entity. The above would assume that the action belongs to the `AfformSubmission` entity. Could we move the action over there and convert it to a use `AbstractBatchAction`? Then the above would know how to handle it, and an added benefit is that the api would be able to process more than one submission at a time.

Reply:
@colemanw 

`process` was added in `Afform` entity as it reuses code in `Civi\Api4\Action\Afform\AbsctractProcessor` which already handles the saving of submitted data.

Moving it to `AfformSubmission` might be bit tricky.

Thoughts?

Reply:
Perhaps we could leave `Afform::process` where it is, but also add an `AfformSubmission::changeStatus` action which is the bulk action for processing 1 or more Afforms (and it delegates work to `Afform::process`. This way the admin could view submissions, select a bunch of checkboxes and bulk process them (or bulk reject them, which might be useful for spam).

Reply:
@colemanw 

Bulk action is already available and working as expected. So admin can process at entry level or via batch.

![Screenshot from 2023-10-25 23-10-56](https://github.com/civicrm/civicrm-core/assets/151481/9bcbcefa-dfac-4e54-97ca-5a2d32ce26e1)

Reply:
I finally figured out how to do this using chaining:
https://github.com/civicrm/civicrm-core/pull/31906